packages/peregrine/lib/talons/Header/useAccountTrigger.js

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useApolloClient, useMutation } from '@apollo/react-hooks';
+
+import { useUserContext } from '@magento/peregrine/lib/context/user';
+import { useDropdown } from '@magento/peregrine/lib/hooks/useDropdown';
+import { clearCartDataFromCache } from '@magento/peregrine/lib/Apollo/clearCartDataFromCache';
+import { clearCustomerDataFromCache } from '@magento/peregrine/lib/Apollo/clearCustomerDataFromCache';
+
+/**
+ *
+ */
+export const useAccountTrigger = props => {
+    const { mutations } = props;
+    const { signOut: signOutMutation } = mutations;
+
+    const {
+        elementRef: accountMenuRef,
+        expanded: accountMenuIsOpen,
+        setExpanded: setAccountMenuIsOpen,
+        triggerRef: accountMenuTriggerRef
+    } = useDropdown();
+
+    const apolloClient = useApolloClient();
+    const history = useHistory();
+    const location = useLocation();
+    const [revokeToken] = useMutation(signOutMutation);
+    const [{ isSignedIn: isUserSignedIn }, { signOut }] = useUserContext();
+
+    const handleSignOut = useCallback(async () => {
+        setAccountMenuIsOpen(false);
+
+        // Delete cart/user data from the redux store.
+        await signOut({ revokeToken });
+        await clearCartDataFromCache(apolloClient);
+        await clearCustomerDataFromCache(apolloClient);
+
+        // Refresh the page as a way to say "re-initialize". An alternative
+        // would be to call apolloClient.resetStore() but that would require
+        // a large refactor.
+        history.go(0);
```

Comment:
How does this compare against `history.push('/')`? `history.go(0)` is gonna refresh the current page, and not take us to the home page. Is that expected?

Reply:
Discussed in Dev Time - this is the desired behavior, yes. We don't always want to send the user back to the home page when they sign out.